react/features/base/react/components/web/Linkify.tsx

```diff
@@ -37,13 +37,42 @@ export default class Linkify extends Component<IProps> {
      * @returns {React$Node}
      */
     _componentDecorator(decoratedHref: string, decoratedText: string, key: number) {
+        let text;
+
+        // In order to prevent homograph attacks we need to use punycode. Reference
+        // https://github.com/tasti/react-linkify/issues/84. In the same time it seems PunycodeJS will treat the URL
+        // as an email when there is '@' and will erase parts of it. This is problematic if there is a URL like
+        // https://example.com/@test@@@123/test@test, punycode will truncate this to https://example.com/@test which
+        // is security issue because parts of the URL are actually missing from the text that we display. That's why
+        // we use punycode on valid URLs(that don't have '@' as part of the host) only for the host part of the URL.
+        try {
+            const url = new URL(decoratedText);
+
+            const { host } = url;
+
+            if (host) {
```

Comment:
I don't think this is possible, the URL would be invalid.

Reply:
It is. Try with `mailto:user@example.com`

Reply:
Ah good one!